### DISCUSSION TITLE
Computational Speed of Conversions

### DISCUSSION THREAD

**Discussion by JoshuaEbenezer** [colour-science/colour#789] [General]
This library is incredible but a lot of functions seem to be really slow. I've been using this for color space or EOTF conversions (BT2020<->BT709 or SMPTE2084<->Linear) and these take a few seconds for 4k images. Do you have any suggestions or plans to improve speed?

**Comment by thomasmansencal** [comment]
Hi @JoshuaEbenezer,

Yes this is acknowledged and last year, one of our GSoC projects was just about that, i.e. The Need for Speed. @OmarWagih1 created a repository to benchmark Colour and implemented support for Cupy in this PR: https://github.com/colour-science/colour/pull/625

It needs to be merged but requires me sourcing a good machine with Cuda.

Cheers,

Thomas

**Reply by JoshuaEbenezer** [reply]
Hi Thomas,

Great, thanks! Will look into the PR.

Regards,
Josh
